src/core/txs/base_tx.c

```diff
@@ -32,11 +32,32 @@ UTXO * dser_UTXO(char * data){
   return new_UTXO;
 }
 
-char * ser_tx(Transaction *tx){
+int size_tx(Transaction *tx){
+    return (sizeof(tx->num_inputs)+sizeof(tx->num_outputs) +
+      tx->num_inputs*sizeof(Input) + tx->num_outputs*sizeof(Output)+sizeof(char));
+}
+
+char * ser_tx(Transaction *tx, char* dest){
+  memcpy(dest, &(tx->num_inputs), sizeof(tx->num_inputs));
+
+  char * num_outputs = dest+sizeof(tx->num_inputs);
+  memcpy(num_outputs, &(tx->num_outputs), sizeof(tx->num_outputs));
+
+  char * inputs = num_outputs+sizeof(tx->num_outputs);
+  memcpy(inputs, tx->inputs, tx->num_inputs*sizeof(Input));
+
+  char * outputs = inputs + tx->num_inputs*sizeof(Input);
+  memcpy(outputs, tx->outputs, tx->num_outputs*sizeof(Output));
+
+  char * terminate = outputs + tx->num_outputs*sizeof(Output);
```

Comment:
Do we need to terminate buffers? It seems like this is only done for transactions?

Reply:
This is indeed a bug. We are not terminating now and are not forced to even if sending on network